magnum_cluster_api/driver.py

```diff
@@ -100,14 +110,26 @@ def update_cluster_status(self, context, cluster, use_admin_ctx=False):
                     ng.destroy()
 
             if cluster.status == "CREATE_IN_PROGRESS":
+                cluster.status_reason = None
                 cluster.status = "CREATE_COMPLETE"
             if cluster.status == "UPDATE_IN_PROGRESS":
+                cluster.status_reason = None
                 cluster.status = "UPDATE_COMPLETE"
 
             cluster.save()
 
         if cluster.status == "DELETE_IN_PROGRESS":
             if capi_cluster and capi_cluster.exists():
+                cluster.status_reason = (
+                    "Cluster status reason: %s. OpenstackCluster status reason: %s"
+                    % (
+                        resources.get_last_cluster_event(self.k8s_api, cluster),
+                        resources.get_last_openstack_cluster_event(
+                            self.k8s_api, cluster
+                        ),
+                    )
+                )
+                cluster.save()
```

Comment:
since this code is repeated, does it make sense to refactor this to a function called `_update_cluster_status` or something along those lines?